File: Bai_102/main.py
def tich_cs(n):
    tich = 1
    while n != 0:
        chu_so = n % 10
        if chu_so != 0: tich *= chu_so
        n //= 10
    return tich

if __name__ == "__main__":
    a, b = map(int, input().split())
    cnt = 0
    for i in range(1, int(b ** 0.5) + 1):
        san_pham = i * tich_cs(i)
        if a <= san_pham < b:
            cnt += 1
    print(cnt) 

File: Bai_102/main.cpp
#include <bits/stdc++.h>
#define ll long long
using namespace std;
ll tich_cs(ll n){
   ll tich = 1;
    while(n != 0){
        ll chu_so = n % 10;
        if(chu_so != 0) tich *= chu_so;
        n /= 10;
    }
    return tich;
}


int main(){
    ll a,b; cin >> a >> b;
    ll dem = 0;
    ll tich, san_pham;
    for(int i = 1; 1LL * i * i < b; ++i){
        san_pham = i * tich_cs(i);
        if(san_pham >= a && san_pham < b)
            dem++; 
    }
    cout << dem;
    return 0;
}